### PR TITLE
Add site-name prefix for multi-site support

### PR DESCRIPTION
**Pull request by dmonty2**
Ability to have multiple schools/cups-servers distinguish between print queues with the same name by adding a site prefix.  e.g.  each "library" queue becomes:
* aberdeen-library
* aeperry-library
* norkam-library

This is similar to Victorealvirus pull request - consider it now re-based.

### CODE REVIEW

**Comment by davesteele** [comment]
The option documented in readme.rst is wrong.

File: cloudprint/cloudprint.py [hunk 1]
```diff
@@ -209,6 +209,8 @@ class CloudPrintProxy(object):
     def __init__(self, auth):
         self.auth = auth
         self.sleeptime = 0
+        self.site = ""
+        self.prefix = False
```
Comment (armooo) [inline cloudprint/cloudprint.py]: Can we use single quotes here for site. We should also be able to use remove the `prefix` and use the fact that an empty string is falsey.

The tests below can be changed from
```
if self.prefix:
```
to 

```
if self.site:
```

**Comment by armooo** [comment]
Can you add a new tests for test_get_printers, add_printer and update_printer with site set you can use the existing tests in test_print_proxy.py as an examples.

The tests can be run locally by installing tox with pip and then running tox in the cloudprint directory.

**Comment by dmonty2** [comment]
I've implemented @davesteele and @armooo suggestions. I'll try to get to the test_print_proxy.py.  Right now we have a working solution for our school district with 50 sites properly whitelisting and prefixing 116 out of 465 printers to the Could.  Thank you for your excellent code base.  I've now been tasked with getting Apple AirPrint working.

**Comment by armooo** [comment]
@dmonty2 thanks for making the updates. I added the new tests committed this.

**Pull request closed without merging**